python/lancedb/table.py

```diff
@@ -697,6 +697,15 @@ def to_arrow(self) -> pa.Table:
         pa.Table"""
         return self._dataset.to_table()
 
+    def to_polars(self) -> "pl.DataFrame":
```

Comment:
It looks like `scan_pyarrow_dataset` returns a `LazyFrame` and not a `DataFrame`